matsumura/AreciboMessage/MessageWriter/Bitmap/PixelIndices.cpp

```diff
@@ -6,47 +6,55 @@ namespace arecibo_message {
 namespace message_writer {
 namespace bitmap {
 
+// ========================================================================================
+// Unnamed Namespace
+// ========================================================================================
+namespace {
+void Repeat(Size loopCount, Function<void(Index)> func) {
```

Comment:
funcの引数がIndexなのでloopCountの型をIndexに変更したほうが適切に思います。
```suggestion
void Repeat(Index loopCount, Function<void(Index)> func) {
```

Reply:
ここは、
* ループカウンターは(VSが提示してくるfor文のsnippetのように)Sizeが良さそう
* funcの引数には頂点のindexを渡して欲しいのでIndexが良い

というの理由で型の選択について悩んだところです。

ちゃんとやるならループカウンターのSize型を引数のIndex型にconvertすべきかと思うのですがいかがでしょう？
で、明示的にconvertするにしても、Size型とIndex型がどちらもstd::size_tのaliasなので単純にキャスト可能なのですがそれは今回/現在たまたまそうなっているだけでは？という点も気色悪かったので、これらのいずれの選択肢も採用せずそのままSize型のループカウンターをIndex型の引数に放り込んでいます。

どうするのがいいんでしょう？

Reply:
size_t(Size)はあらゆる型のオブジェクトの最大サイズということなので、
ループカウンタには基本的にSizeを用いるのがいいのかなと思います。

明示的にconvertについては、私も悩んだ結果、
今回の場合は配列などを使っていないのと、
コード自体が短いのでループカウンタの型もIndexでも良いと思いましたが、
最近はVSのintelligence等で、同じ型が元になっていることが簡単にわかると思うので、
このままでいいと思います。

同じ型のaliasがそれぞれ別の型に認識される方法がないのかと、
調べているとopacue alias(strong typedef)についての記事があったので貼っておきます。
残念ながらデフォルトの機能としては無いようでしたが...
https://cpplover.blogspot.com/2013/09/c1yopaque-alias.html

Reply:
『寝言です』が寝言でした。まさにご指摘の部分で悩んでいました。
迷走してすみません。

ここの処理の意味合いですが、 `Repeat()` の呼び出し側としては
* `func` で指定する処理を `scale` 回繰り返してもらいたい。
* `func` の引数に渡される `今何回目なのか` という情報を基に、何らかの処理をしたい。現状、以下の2例。
  * 処理対象の行番号(Index型)を算出したい。
  * `indices` に追加するピクセル番号(Index型)を算出したい。

となります。

他方、呼び出される `Repeat()` 側としては
* `func` で指定される何らかの処理を、 `loopCount` で指定された回数繰り返す。
* 今何回目なのかを `func` に伝えてあげる。それを使って呼び出し側が何をしたいのかは知らない。

ということになります。

従って、
* `func` の引数に渡されるのは `今何回目なのか` なので、引数の型をSize型にする。
* `Repeat()` 呼び出し側で、 `今何回目なのか` をSize型から自分が使いたい型(今回は *たまたま* どちらもIndex型)に変換して利用する。

が良いように思います。

その上で、
### 今回の実装では(偶然ではあるが)SizeとIndexはどちらもstd::size_t型だった。
ということを鑑み、
### 特に明示的に変換する処理は書かず、そのまま利用する。
という理屈でいかがでしょう？

P.S.
> 調べているとopacue alias(strong typedef)についての記事があったので貼っておきます。

出た、本の虫！！
OOP 9Rulesの「primitive型のwrap」とかで使えそうですね。(C++1yで検討中とのことなので、C++20には入ったのかな？)
あと、不透過は opaque ですね。GUIプログラムをやってるとよく見かける単語です。

Reply:
1d9e215 で対応しました。